### DISCUSSION TITLE
add more positions for signals to curved rails

### DISCUSSION THREAD

**Discussion by L0ric0** [voxeltycoon/issues#915] [General]
Lets consider a railsystem that works on two one way tracks (right hand drive). And lets consider exit from a track to a station.

![bug_voxeltycoon_2](https://user-images.githubusercontent.com/22472023/120923918-59ae6d00-c6d1-11eb-9814-62bc2013b3fb.png)

the left exit is an optimal one where two trains going in different directions can pass each other without one of them stopping at the Pre-Signal at the entrance of the interaction. The problem with that system is, that the intersection takes way to much space (it is as wide as a 4 rail station) for what it does, but it is impossible to not use this type of intersection on somewhat dense networks.

On the right side is a system wich is only as wide as a 2 rail station but has the  problem that one of the trains going throu this intersections has to stop and wait for the other.

This problem could be solved by adding more places for railsignals on curved rails. this places would not need to be available when the default places are available.

**Comment by Dojando1** [comment]
I would say that is a feature request but I agree! The problem could be solved with the infamous path signal from open-ttd. 
That would let a train go into a block if the pre determined path will not collide with another train in the same block. So a train could go into the station while another train is leaving it without pressing a signal in the tiny spot between those two tracks which would not just look ugly but would be unrealistic as well.

**Comment by xmnovotny** [comment]
You can use [Path signal mod](https://steamcommunity.com/sharedfiles/filedetails/?id=2565417798) and that intersection will be fine...

**Comment by Dojando1** [comment]
> You can use [Path signal mod](https://steamcommunity.com/sharedfiles/filedetails/?id=2565417798) and that intersection will be fine...

Nice, thank you! :)
I didn't know this existed.

**Comment by ArielCMathews** [comment]
I agree, as an alternative fix, maybe they could make it so that the signals are placed on the tracks themselves rather than needing space next to the tracks, so that we could have closer passing intersections and the signals would still place where they should be. It's such a pain to have to make big sprawling intersections when a tidy triangle would do, with the ability to signal properly in smaller spaces.
